### ISSUE TITLE
ReceiveAsync does not throw OperationCancelledException but IotHubCommunicationException upon cancellation

### ISSUE THREAD

**Issue by sensslen** [Azure/azure-sdk-for-net#25643]
When cancelling a ReceiveAsync(cancellationToken) call using the cancellation token, it throws an instance of IotHubCommunicationException instead of an instance of OperationCancelledException as documented

Feel free to use the following code snippet to reproduce the issue:
```csharp
using System.Text;
using Microsoft.Azure.Devices.Client;

var cancellationTokenSource = new CancellationTokenSource();
using var exitEvent = new AutoResetEvent(false);

Console.CancelKeyPress += (sender, args) =>
{
    Console.WriteLine("Request cancellation");

    cancellationTokenSource.Cancel();
    exitEvent.WaitOne();
    args.Cancel = true;
};

var url = "url";
var deviceId = "deviceId";
var symmetricKey = "key";
var authentication = new DeviceAuthenticationWithRegistrySymmetricKey(deviceId, symmetricKey);
using var client = DeviceClient.Create(url, authentication, TransportType.Amqp);
client.SetRetryPolicy(new ExponentialBackoff(10, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(10),
    TimeSpan.FromSeconds(2)));

var receptionTask = Reception(client, cancellationTokenSource.Token);

receptionTask.Wait();
Console.WriteLine("finished");
exitEvent.Set();

async Task Reception(DeviceClient client, CancellationToken cancellationToken)
{
    while (!cancellationToken.IsCancellationRequested)
    {
        try
        {
            using var message = await client.ReceiveAsync(cancellationToken);
            var messageBytes = message.GetBytes();
            Console.WriteLine($"Received:{Encoding.UTF8.GetString(messageBytes)}");
            await client.CompleteAsync(message, cancellationToken);
        }
        catch (OperationCanceledException)
        {
            // exepcted as of documenation
        }
        catch (Exception e)
        {
            // unexpected error
            Console.WriteLine(e);
        }
    }
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 15ca5154-e000-ef77-9436-c21df51f37b5
* Version Independent ID: 6970ce6d-a195-93f7-e165-c30820773b01
* Content: [DeviceClient.ReceiveAsync Method (Microsoft.Azure.Devices.Client) - Azure for .NET Developers](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.deviceclient.receiveasync?view=azure-dotnet#Microsoft_Azure_Devices_Client_DeviceClient_ReceiveAsync_System_Threading_CancellationToken_)
* Content Source: [xml/Microsoft.Azure.Devices.Client/DeviceClient.xml](https://github.com/Azure/azure-docs-sdk-dotnet/blob/master/xml/Microsoft.Azure.Devices.Client/DeviceClient.xml)
* Service: **iot-hub**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by PramodValavala-MSFT** [comment]
@sensslen Looks like this is part of the [Device SDK code](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/iothub/device/src/InternalClient.cs#L1296) and I also see you have another issue in the respective repo already - Azure/azure-iot-sdk-csharp#2246

Since this is already being addressed there, closing this one in favor of that.

**Comment by sensslen** [comment]
The issue mentioned is regarding cancellation. this issue is regarding the exception thrown. Even though they are somewhat related, they are different. The documentation page does state OperationCancelled exception to be thrown. Even with the fix in Azure/azure-iot-sdk-csharp#2246 the exception is still IotHubCommunicationException.
